src/screencast/pipewire_screencast.c

```diff
@@ -163,10 +157,75 @@ static void pwr_handle_stream_param_changed(void *data, uint32_t id,
 	pw_stream_update_params(stream, params, 2);
 }
 
+static void pwr_handle_stream_add_buffer(void *data, struct pw_buffer *buffer) {
+	struct xdpw_screencast_instance *cast = data;
+	struct spa_data *d;
+
+	logprint(TRACE, "pipewire: add buffer event handle");
+
+	d = buffer->buffer->datas;
+
+	// Select buffer type from negotiation result
+	if ((d[0].type & (1u << SPA_DATA_MemFd)) > 0) {
+		d[0].type = SPA_DATA_MemFd;
+	} else {
+		logprint(ERROR, "pipewire: unsupported buffer type");
+		cast->err = 1;
+		return;
+	}
+
+	logprint(TRACE, "pipewire: selected buffertype %u", d[0].type);
+	// Prepare buffer for choosen type
+	if (d[0].type == SPA_DATA_MemFd) {
+		d[0].maxsize = cast->simple_frame.size;
+		d[0].mapoffset = 0;
+		d[0].chunk->size = cast->simple_frame.size;
+		d[0].chunk->stride = cast->simple_frame.stride;
+		d[0].chunk->offset = 0;
+		d[0].flags = 0;
+		d[0].fd = anonymous_shm_open();
+
+		if (d[0].fd == -1) {
+			logprint(ERROR, "pipewire: unable to create anonymous filedescriptor");
+			return;
```

Comment:
Should we set `cast->err` for this error condition? (And the next 2 as well)

Reply:
yeah sounds good. Don't know if we should check for that before registering the wlroots callback or somewhere else.

Reply:
Hm, yeah we should probably check this as soon as we get the broken buffer from somewhere else, which is probably when sending the screencopy request to the compositor. It's unfortunate that `add_buffer` can't return an error.